### ISSUE TITLE
Add use_github_action

### ISSUE THREAD

**Issue by annakrystalli** [benmarwick/rrtools#132]
**Please wait for some discussion of your request before making a Pull Request.** 

**Is your feature request related to a problem? Please describe.**

Many are using GitHub Actions for CI. For researchers, the ability to test on many systems with a single CI framework and script could lower the barrier to understanding their codes behaviour on different systems.

**Describe the solution you'd like**

Create a new `use_github_action` function based on `usethis::use_github_action` (I believe still only available in developing) (perhaps with a more limited build matrix by default) and translate the Travis build workflow.

** Describe alternatives you've considered**

**Additional context**

**Comment by benmarwick** [comment]
I agree this sounds like a very good idea that we should implement. @annakrystalli would you like to make a PR with an `rtools::use_github_action` that is in the spirit of our current `rtools::use_travis` that includes the docker container?

**Comment by annakrystalli** [comment]
I could definitely give it a go!

It might be a couple of weeks till I can get to it though.

**Comment by nevrome** [comment]
Excellent! There is some interesting code in #109 that could serve as an inspiration. I think Github and Gitlab use a similar setup for their CI tools?

**Comment by benmarwick** [comment]
I think this is the way to go. Use of travis and circle is now very uncommon in the R community, with most people adopting GitHub Actions for CI. I've been using GitHub Actions on my research compendia for a while and it's been fine. I propose to remove the functions for using travis and circle from rrtools, and add a function for GitHub Actions. What do you all reckon?

**Comment by wolass** [comment]
I have to say that I really rarely used Travis/ circle ci functions for my
compendia, probably due to lack of proper  knowledge how to use them.

If you guys think that github actions have a lower barrier of adoption a
will gladly give it a go!



Dr. med. Wojciech Francuzik <http://fracoscientific.com/>
Assistenzarzt
Allergologie und Immunologie
Klinik für Dermatologie, Venerologie und Allergologie
Campus Charite Mitte, Universitätsmedizin Berlin
Charitéplatz 1, 10117 Berlin
Tel +49 (30) 450 618 344


śr., 3 sie 2022 o 00:18 Ben Marwick ***@***.***> napisał(a):
